test/functional/rpc_setban.py

```diff
@@ -42,5 +42,15 @@ def run_test(self):
         peerinfo = self.nodes[1].getpeerinfo()[0]
         assert(not 'noban' in peerinfo['permissions'])
 
+        tor_addr = "pg6mmjiyjmcrsslvykfwnntlaru7p5svn6y2ymmju6nubxndf4pscryd.onion"
+        self.nodes[1].setban(tor_addr, "add")
+        is_banned = False
+        for ban_entry in self.nodes[1].listbanned():
+            if ban_entry['address'] == tor_addr:
+                is_banned = True
+                break
+        assert(is_banned)
+        self.nodes[1].setban(tor_addr, "remove")
```

Comment:
Here the test sees only one ban, the added one. If it is moved above the previous test, there will also be a second ban in listbanned, e.g.
```python3
[
  {'address': '127.0.0.1/32', ...},
  {'address': 'pg6mmjiyjmcrsslvykfwnntlaru7p5svn6y2ymmju6nubxndf4pscryd.onion', ...}
]
```

Here's a suggestion to do that, add logging, and also assert pre-add and post-remove:
```diff
 +TOR_ADDR = "pg6mmjiyjmcrsslvykfwnntlaru7p5svn6y2ymmju6nubxndf4pscryd.onion"
+
 class SetBanTests(BitcoinTestFramework):
     def set_test_params(self):
         self.num_nodes = 2
         self.setup_clean_chain = True
         self.extra_args = [[],[]]
 
+    def banned_tor_addr(self, tor_addr=TOR_ADDR):
+        return list(filter(lambda ban_entry: ban_entry["address"] == tor_addr, self.nodes[1].listbanned()))
+
     def run_test(self):
         # Node 0 connects to Node 1, check that the noban permission is not granted
         self.connect_nodes(0, 1)
@@ -35,6 +40,13 @@ class SetBanTests(BitcoinTestFramework):
         peerinfo = self.nodes[1].getpeerinfo()[0]
         assert('noban' in peerinfo['permissions'])
 
+        self.log.info("Test that a Tor v3 address can be banned/unbanned")
+        assert(not self.banned_tor_addr())
+        self.nodes[1].setban(TOR_ADDR, "add")
+        assert(self.banned_tor_addr())
+        self.nodes[1].setban(TOR_ADDR, "remove")
+        assert(not self.banned_tor_addr())
+
         # If we remove the ban, Node 0 should be able to reconnect even without noban permission
         self.nodes[1].setban("127.0.0.1", "remove")
         self.restart_node(1, [])
@@ -42,15 +54,5 @@ class SetBanTests(BitcoinTestFramework):
         peerinfo = self.nodes[1].getpeerinfo()[0]
         assert(not 'noban' in peerinfo['permissions'])
 
-        tor_addr = "pg6mmjiyjmcrsslvykfwnntlaru7p5svn6y2ymmju6nubxndf4pscryd.onion"
-        self.nodes[1].setban(tor_addr, "add")
-        is_banned = False
-        for ban_entry in self.nodes[1].listbanned():
-            if ban_entry['address'] == tor_addr:
-                is_banned = True
-                break
-        assert(is_banned)
-        self.nodes[1].setban(tor_addr, "remove")
-
 if __name__ == '__main__':
```

Reply:
Added something like that in 39b43298d, thanks!